include/fox/inplace_free_list.hpp

```diff
@@ -127,13 +132,106 @@ namespace fox
 			return first_free_ == offset_type_npos;
 		}
 
+		[[nodiscard]] bool is_free_list_sorted() const
+		{
+			offset_type current = first_free_;
+			while (current != offset_type_npos)
+			{
+				offset_type next = reinterpret_cast<const offset_accessor*>(std::data(storage_))[current].offset;
+				if (next <= current) return false;
+				current = next;
+			}
+			return true;
+		}
+
 	public:
 		void clear() noexcept
 		{
 			destroy_all();
 			initialize_empty();
 		}
 
+		template<std::invocable<pointer, pointer> Callback>
+		void optimize(Callback&& cb)
+		{
+			optimize_at([&](size_type from, size_type to) { cb(this->data() + from, this->data() + to); });
+		}
+
+		template<std::invocable<size_type, size_type> Callback>
+		void optimize_at(Callback&& cb)
+		{
+			std::bitset<Capacity> mask = free_mask();
+			offset_type left = 0;
+			offset_type right = static_cast<offset_type>(Capacity - 1);
+			while (left < right)
+			{
+				if (mask.test(left))
+				{
+					if (mask.test(right))
+					{
+						reinterpret_cast<offset_accessor*>(std::data(storage_))[right].offset = right + 1;
+						right--;
+					}
+					else
+					{
+						T* source = reinterpret_cast<T*>(std::data(storage_)) + right;
+						T* dest = reinterpret_cast<T*>(std::data(storage_)) + left;
+						std::construct_at(dest, std::move(*source));
```

Comment:
You are not calling `std::destroy_at` after the move and before the construction of the offset accessor.